design-editor/src/utils/path-utils.js

```diff
@@ -60,6 +61,53 @@ export default {
 	 */
 	getFileName(pathName, ext) {
 		return ext ? path.basename(pathName) : path.basename(pathName, path.extname(pathName));
+	},
+
+	/**
+	 * Set a source to background image or image
+	 * @param {Event} event
+	 * @param {string} changedAttribute name of changed attribute
+	 * @param {string} elementId id of changed element
+	 * @param {function} changeImageInfo function that changed info about element in sidebar
+	 */
+	setImageSource(event, changedAttribute, elementId, changeImageInfo) {
+		[].slice.call(event.target.files).forEach(file => {
+			const reader = new FileReader();
+			reader.addEventListener('loadend', event => {
+				if (event.target.readyState === FileReader.DONE) {
+					const
+						filePath = path.join('images', file.name),
+						writePath = this.createProjectPath(filePath),
+						readPath = this.createProjectPath(filePath, true);
+
+					utils.checkGlobalContext('writeFile') (
+						writePath,
+						event.target.result, {
+							encoding: 'binary'
+						},
+						() => {
+							if (changedAttribute === 'src') {
+								AppManager.getActiveDesignEditor()
+									.getModel().updateAttribute(
```

Comment:
I'd move `AppManager.getActiveDesignEditor().getModel()` outside if (just before it) and assign to a variable as it repeats both in `if` and in `else` block. It should be more readable this way : )